prereise/gather/griddata/transmission/geometry.py

```diff
@@ -250,6 +251,58 @@ def calculate_shunt_capacitance_per_km(self):
         return capacitance_per_km
 
 
+@dataclass
+class Line(DataclassWithValidation):
+    """Given a Tower design, line voltage, and length, calculate whole-line impedances
+    and rating.
+
+    :param Tower tower: tower parameters (containing per-kilometer impedances).
+    :param int/float length: line length (kilometers).
+    :param int/float voltage: line voltage (kilovolts).
+    :param int/float freq: the system nominal frequency (Hz).
+    """
+
+    tower: Tower
+    length: float
+    voltage: float
+    freq: float = 60.0
+    series_impedance_per_km: complex = field(init=False)
+    shunt_admittance_per_km: complex = field(init=False)
+    propogation_constant_per_km: complex = field(init=False)
+    surge_impedance: complex = field(init=False)
+    series_impedance: complex = field(init=False)
+    shunt_admittance: complex = field(init=False)
+
+    def __post_init__(self):
+        # Convert integers to floats as necessary
+        for attr in ("freq", "length", "voltage"):
+            if isinstance(getattr(self, attr), int):
+                setattr(self, attr, float(getattr(self, attr)))
```

Comment:
For simplicity, maybe we can cast everything into `float` regardlessly. The current implementation is good to go as well.

Reply:
The intention here is to be a little flexible about the inputs before the validation method, but not too flexible. Currently, if we try to pass e.g. a `str` as any of these inputs, the `validate_input_types` method will catch it and return an informative error: `TypeError: for voltage: expected <class 'float'>, but got <class 'str'>`. If we remove the `if` block, the error is less informative: `ValueError: could not convert string to float: 'a'`.

Reply:
I've been exploring how to support union types in this situation which also works with the validation, but got  `TypeError: Subscripted generics cannot be used with class and instance checks` when using the `Union[int, float]` trick. Let's keep it as it is then.